kubernetes.go

```diff
@@ -45,7 +46,7 @@ type KubeController struct {
 	hasSynced   bool
 }
 
-func newKubeController(ctx context.Context, c *kubernetes.Clientset, gw *gatewayClient.Clientset) *KubeController {
+func newKubeController(ctx context.Context, c *kubernetes.Clientset, gw *gatewayClient.Clientset, filters ResourceFilters) *KubeController {
```

Comment:
there might be a better way to do this rather than passing it through the kubeController function.
it should probably be directly read in the `newKubeController` function

Reply:
ignore my comment here, this is fine it's attached to each "controller"